external/fv3fit/fv3fit/keras/_models/shared/pure_keras.py

```diff
@@ -164,3 +165,23 @@ def dump(self, path: str) -> None:
                         }
                     )
                 )
+
+    def _renamed_dict_compatible_model(self) -> tf.keras.Model:
```

Comment:
This looks quite similar to some of the code here: https://github.com/ai2cm/fv3net/blob/0d539792b7ee9804e47b0cdb684e1e03682fe5d8/external/fv3fit/fv3fit/keras/adapters.py#L21

Can you re-use some of that code? The tests for `self.dict_compatible_model` are also quite long and similar to the tests here: https://github.com/ai2cm/fv3net/blob/0d539792b7ee9804e47b0cdb684e1e03682fe5d8/external/fv3fit/tests/keras/test_adapters.py#L48

If you added an `adapters.rename_inputs` function  to go along with `rename_dict_output`, then you could implement this function like this:
```
rename_dict_outputs(outputs(rename_inputs(model, self.input_variables), self.output_variables))
```

At that point, there is probably no need for any special tests...mypy is probably enough to ensure that `self.dict_compatible_model` is a keras model.